doc/guide/authentication.xml

```diff
@@ -99,4 +99,49 @@
 
     </section>
   </section>
+
+  <section id="2fa-auth">
+    <title>Enable two-factor authentication</title>
+
+    <para>As with any other <code>pam</pam> based service that supports it,
+      you can harden your Cockpit authentication by adding two-factor authentication.
+      This can be achieved by installing the <code>google-authenticator</code> package
+      and configuring <code>pam</code> to use it with Cockpit.</para>
+
+    <para>As the user you want to set up two-factor auth for, configure
+      <code>google-authenticator</code> as per its <ulink url="https://github.com/google/google-authenticator-libpam">documentation</ulink>.
+      Then, edit the <filename>/etc/pam.d/cockpit</filename> file and add the following:
+    </para>
+
+<programlisting>
+auth       required     pam_google_authenticator.so
```

Comment:
Add this where?

Also: is this the same across multiple distros?

Does this work in the various "common" files as well, if we want this to work for all system services?

Reply:
The config line depends on pam so it should be the same.
About the other services, as we setup two-factor auth specifically for Cockpit in the `/etc/pam.d/cockpit` it does not interfere with other settings.